### ISSUE TITLE
Remove #![feature(const_generics)]?

### ISSUE THREAD

**Issue by chubei** [RSSchermer/std140.rs#1]
Now that const generics has been [stabilized](https://blog.rust-lang.org/2021/03/25/Rust-1.51.0.html#const-generics-mvp), can we consider removing `#![feature(const_generics)]` and make this crate work on stable?

**Comment by RSSchermer** [comment]
Good point! I've removed the `const_generics` feature dependency.

There turned out to be a new feature dependency on `const_fn`, presumably introduced by ongoing work on the [const fn bounds RFC](https://github.com/oli-obk/rfcs/blob/const_generic_const_fn_bounds/text/0000-const-generic-const-fn-bounds.md). I've changed the way the `repr_std140` attribute macro typechecks struct fields to resolve this, which results in somewhat more verbose error messages if a struct field is not std140 compatible. However, that seems an acceptable trade-off for getting this crate to compile on stable, so I've just published version `0.2.3` which should compile correctly on Rust stable `1.51`.

I'll look into reverting to the other way to typecheck struct fields  - trait bounds on a `const fn` - once stable Rust allows this. If you're reading this in the future and trait bounds on `const fn`s have been stabilized, please feel free to ping me about this.

**Comment by chubei-oppen** [comment]
The new typecheck seems to be breaking when there're multiple fields with the same type. For example

```rust
#[std140::repr_std140]
struct Uniform {
    a: std140::float,
    b: std140::float,
}
```

is not compiling.

**Comment by chubei** [comment]
The new typecheck seems to be breaking when there're multiple fields with the same type. For exmaple

```rust
#[std140::repr_std140]
struct Uniform {
    a: std140::float,
    b: std140::float,
}
```

is not compiling.

**Comment by RSSchermer** [comment]
In hindsight, that was obviously going to be an issue with the new typechecking method... I've switched to yet another method and added some compilation tests so that this shouldn't go unnoticed in the future. Should be resolved in `0.2.4`.

Thanks!

**Comment by chubei** [comment]
Thank YOU! Your response has been incredibly fast.
